apm-agent-core/src/main/java/co/elastic/apm/agent/tracemethods/TraceMethodInstrumentation.java

```diff
@@ -146,6 +146,10 @@ public static Object onMethodEnter(@Advice.Origin Class<?> clazz,
                     span.withName(signature).activate();
                 }
             } else if (parent.isSampled()) {
+                if (parent.shouldSkipChildSpanCreation()) {
+                    // span limit reached means span will not be reported, thus we can optimize and skip creating one
+                    return null;
```

Comment:
You have a debug-log for this case in the other instrumentation, I think it would make sense to add a debug log here aswell.